File: mobile/lib/models/cart_item.dart
import 'package:mobile/models/product.dart';
import 'package:mobile/models/product_variation.dart';

class CartItem {
  final Product product;
  final ProductVariation? variation;
  int quantity;
  double discount;
  String? notes;

  CartItem({required this.product, this.variation, this.quantity = 1, this.discount = 0.0, this.notes});

  double get price => variation?.price ?? product.price;

  double get total => (price - discount) * quantity;
}

File: desktop/Models/CartItem.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace KioslyDesktop.Models
{
    public class CartItem : INotifyPropertyChanged
    {
        private int _quantity;

        public Product Product { get; set; }

        public int Quantity
        {
            get => _quantity;
            set
            {
                if (_quantity != value)
                {
                    _quantity = value;
                    OnPropertyChanged();
                    OnPropertyChanged(nameof(Total));
                }
            }
        }

        public double Total => Product.Price * Quantity;

        public event PropertyChangedEventHandler? PropertyChanged;

        protected void OnPropertyChanged([CallerMemberName] string? propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}
